realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsManagedTests.java

```diff
@@ -0,0 +1,423 @@
+    @Test
+    @Ignore
+    public void queryStartingWithBacklink() {
+        realm.beginTransaction();
+        AllJavaTypes gen1 = realm.createObject(AllJavaTypes.class, 10);
+
+        AllJavaTypes gen2A = realm.createObject(AllJavaTypes.class, 1);
+        gen2A.setFieldObject(gen1);
+
+        AllJavaTypes gen2B = realm.createObject(AllJavaTypes.class, 2);
+        gen2B.setFieldObject(gen1);
+
+        AllJavaTypes gen3 = realm.createObject(AllJavaTypes.class, 3);
+        RealmList<AllJavaTypes> parents = gen3.getFieldList();
+        parents.add(gen2A);
+        parents.add(gen2B);
+
+        realm.commitTransaction();
+
+        RealmResults<AllJavaTypes> result = realm.where(AllJavaTypes.class)
+            .greaterThan("objectParents.fieldId", 1)
+            .findAll();
+        assertEquals(1, result.size());
+        assertTrue(result.contains(gen2B));
+    }
+
+    // Query on a field descriptor that ends with a backlink
+    @Test
+    @Ignore
+    public void queryEndingWithBacklink() {
+        realm.beginTransaction();
+        AllJavaTypes gen1 = realm.createObject(AllJavaTypes.class, 10);
+
+        AllJavaTypes gen2A = realm.createObject(AllJavaTypes.class, 1);
+        gen2A.setFieldObject(gen1);
+
+        AllJavaTypes gen2B = realm.createObject(AllJavaTypes.class, 2);
+        gen2B.setFieldObject(gen1);
+
+        AllJavaTypes gen3 = realm.createObject(AllJavaTypes.class, 3);
+        RealmList<AllJavaTypes> parents = gen3.getFieldList();
+        parents.add(gen2A);
+        parents.add(gen2B);
+
+        realm.commitTransaction();
+
+        RealmResults<AllJavaTypes> result = realm.where(AllJavaTypes.class)
+            .isNotNull("objectParents.listParents")
+            .findAll();
+        assertEquals(2, result.size());
+        assertTrue(result.contains(gen2A));
+        assertTrue(result.contains(gen2B));
+    }
+
+    // Query on a field descriptor that has a backlink in the middle
+    @Test
+    @Ignore
+    public void queryBacklinkInMiddle() {
+        realm.beginTransaction();
+        AllJavaTypes gen1 = realm.createObject(AllJavaTypes.class, 10);
+
+        AllJavaTypes gen2A = realm.createObject(AllJavaTypes.class, 1);
+        gen2A.setFieldObject(gen1);
+
+        AllJavaTypes gen2B = realm.createObject(AllJavaTypes.class, 2);
+        gen2B.setFieldObject(gen1);
+
+        AllJavaTypes gen3 = realm.createObject(AllJavaTypes.class, 3);
+        RealmList<AllJavaTypes> parents = gen3.getFieldList();
+        parents.add(gen2A);
+        parents.add(gen2B);
+
+        realm.commitTransaction();
+
+        RealmResults<AllJavaTypes> result = realm.where(AllJavaTypes.class)
+            .lessThan("objectParents.listParents.fieldId", 4)
+            .findAll();
+        assertEquals(2, result.size());
+    }
+
+    // A newly added notification callback should be called immediately for an object
+    // that has acquired a backlink
+    // !!!FIXME What does this test?
```

Comment:
My initial test list was probably incorrect in places because I didn't understand everything yet, so I wouldn't take everything it said as the truth 😄 
I would probably just delete it then.